Makefile

```diff
@@ -276,81 +259,96 @@ proto-lint: $(BUF)
+	@mkdir -p $(CACHE_INCLUDE)
+	@mkdir -p $(CACHE_VERSIONS)
 
-	mkdir -p $(SDK_QUERY_TYPES)
-	curl -sSL $(COSMOS_SDK_PROTO_URL)/query/v1beta1/pagination.proto > $(SDK_QUERY_TYPES)/pagination.proto
+$(GOLANGCI_LINT_VERSION_FILE): $(CACHE)
+	@echo "installing golangci-lint..."
+	@rm -f $(CACHE_BIN)/golangci-lint
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
+		sh -s -- -b $(CACHE_BIN) $(GOLANGCI_LINT_VERSION)
+	@rm -rf $(dir $@)
+	@mkdir -p $(dir $@)
+	@touch $@
 
-	mkdir -p $(SDK_COIN_TYPES)
-	curl -sSL $(COSMOS_SDK_PROTO_URL)/v1beta1/coin.proto > $(SDK_COIN_TYPES)/coin.proto
+$(GOLANGCI_LINT): $(GOLANGCI_LINT_VERSION_FILE)
 
-cache-setup:
-	@mkdir -p $(CACHE_BIN)
-	@mkdir -p $(CACHE_INCLUDE)
+.PHONY:lintdeps-install
+lintdeps-install: $(GOLANGCI_LINT)
+	@echo "lintdeps-install is deprecated and will be removed once Github Actions migrated to use .cache/bin/golangci-lint"
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
+		sh -s -- -b $(GOBIN) $(GOLANGCI_LINT_VERSION)
 
-$(BUF):
-	@echo "Installing protoc buf cli..."
-	@rm -f $@
+$(BUF_VERSION_FILE): $(CACHE)
+	@echo "installing protoc buf cli..."
+	@rm -f $(CACHE_BIN)/buf
 	@curl -sSL \
 		"https://github.com/bufbuild/buf/releases/download/v$(BUF_VERSION)/buf-$(UNAME_OS)-$(UNAME_ARCH)" \
 		-o "$(CACHE_BIN)/buf"
 	@chmod +x "$(CACHE_BIN)/buf"
-
-$(PROTOC):
-	@echo "Installing protoc compiler..."
-	@rm -f $@
+	@rm -rf $(dir $@)
+	@mkdir -p $(dir $@)
+	@touch $@
+$(BUF): $(BUF_VERSION_FILE)
+
+$(PROTOC_VERSION_FILE): $(CACHE)
+	@echo "installing protoc compiler..."
+	@rm -f $(CACHE_BIN)/protoc
 	@(cd /tmp; \
 	curl -sOL "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/${PROTOC_ZIP}"; \
 	unzip -oq ${PROTOC_ZIP} -d $(CACHE) bin/protoc; \
 	unzip -oq ${PROTOC_ZIP} -d $(CACHE) 'include/*'; \
 	rm -f ${PROTOC_ZIP})
+	@rm -rf $(dir $@)
+	@mkdir -p $(dir $@)
+	@touch $@
+$(PROTOC): $(PROTOC_VERSION_FILE)
 
-.PHONY: proto-tools
-proto-tools: cache-setup $(BUF) $(PROTOC)
+$(MODVENDOR): $(CACHE)
+	@echo "installing modvendor..."
+	GOBIN=$(CACHE_BIN) GO111MODULE=off go get github.com/goware/modvendor
 
-tools-clean:
+kubetypes-deps-install:
+	if [ -d "$(shell go env GOPATH)/src/k8s.io/code-generator" ]; then    \
+		cd "$(shell go env GOPATH)/src/k8s.io/code-generator" && git pull;  \
+		exit 0;                                                             \
+	fi;                                                                   \
+	mkdir -p "$(shell go env GOPATH)/src/k8s.io" && \
+	git clone  git@github.com:kubernetes/code-generator.git \
+		"$(shell go env GOPATH)/src/k8s.io/code-generator"
+
+devdeps-install: $(GOLANGCI_LINT) kubetypes-deps-install
+	$(GO) install github.com/vektra/mockery/.../
+	$(GO) install k8s.io/code-generator/...
+	$(GO) install sigs.k8s.io/kind
+	$(GO) install golang.org/x/tools/cmd/stringer
+	GOBIN=$(CACHE_BIN) GO111MODULE=off go get github.com/goware/modvendor
```

Comment:
handled by `$(MODVENDOR)` target?  can we not use `tools.go` for this?  does it have to be installed in `$(CACHE_BIN)`?

Reply:
copy-paste at 1am